tests/test_radiocard.py

```diff
@@ -0,0 +1,71 @@
+from dash import Dash, html, Output, Input, _dash_renderer
+import dash_mantine_components as dmc
+
+_dash_renderer._set_react_version("18.2.0")
+
+
+def radiogroup_app(**kwargs):
+    app = Dash(__name__)
+
+    app.layout = dmc.MantineProvider(
+        html.Div(
+            [
+                dmc.RadioGroup(
+                    id="radio-group",
+                    children=dmc.Group(
+                        [
+                            dmc.RadioCard(value="option1", children=dmc.RadioIndicator(), id="o1"),
+                            dmc.RadioCard(value="option2",  id="o2"),
+                            dmc.RadioCard(value="option3", children=dmc.Text("Text"),  id="o3"),
+                        ]
+                    ),
+                    **kwargs,
+                ),
+                html.Div(id="output"),
+            ]
+        )
+    )
+
+    @app.callback(Output("output", "children"), Input("radio-group", "value"))
+    def update_output(selected_values):
+        return f"Selected: {selected_values}"
+
+    return app
+
+
+def test_001rc_radio_group(dash_duo):
+
+    app = radiogroup_app()
+    dash_duo.start_server(app)
+
+    # Wait for the app to load
+    dash_duo.wait_for_element("div[aria-labelledby='radio-group-label']")
+
+    option1 = dash_duo.find_element("#o1")
+    option2 = dash_duo.find_element("#o2")
+    option1.click()
+    dash_duo.wait_for_text_to_equal("#output", "Selected: option1")
+    option2.click()
+    dash_duo.wait_for_text_to_equal("#output", "Selected: option2")
+    # Not deselectable
+    option2.click()
+    dash_duo.wait_for_text_to_equal("#output", "Selected: option2")
```

Comment:
This one, on the other hand, seems like it needs a `sleep` that it doesn't have. The only way I can think of to test this without a `sleep` would be to put a [call count](https://github.com/plotly/dash/blob/2bd32585b4874bc85bd83c9a4b38af657c23c414/tests/integration/callbacks/test_basic_callback.py#L91) in the callback, do something after the second `option2.click()` to cause another change, like clicking back to option1 and verifying that so you know it's done, and then checking the call count to ensure that the callback didn't fire on the click that didn't deselect option2.

That's a bit cumbersome, so if you'd prefer to just add in a `sleep` that's OK too.

Reply:
changed with https://github.com/snehilvj/dash-mantine-components/pull/509/commits/4df621501cca4c75631dc2deea7496baee720bd3